config/config.go

```diff
@@ -87,8 +89,16 @@ type azureKVKey struct {
 	Version  string `yaml:"version"`
 }
 
+type destinationRule struct {
+	PathRegex        string       `yaml:"path_regex"`
+	S3Bucket         string       `yaml:"s3_bucket"`
+	S3Prefix         string       `yaml:"s3_prefix"`
+	GCSBucket        string       `yaml:"gcs_bucket"`
+	GCSPrefix        string       `yaml:"gcs_prefix"`
```

Comment:
would it be more elegant to use a single variable to store the target, then switch on the protocol method? It would allow us to add new publication protocols without changing this struct.

Reply:
So something like?:
```
destination_rules:
  - location: "s3://my-bucket/secrets"
```

That could work, but could become weird if certain publish destinations don't map well to that. For example, could we specify all of the needed parameters for a Vault destination in a single variable? I don't think it's a problem to have to make additions to the config format to support a new destination?

Reply:
if it's a string, it could also be `vault_user=foo vault_pass=bar vault_url=http://example.net`

Reply:
True, but then we will have to support more and more complex parsing logical and will be forced to pack more and more options into a single line rather than various kv pairs.

imo before we ship this feature, it would be worth getting more destinations supported (k8s secrets, vault, etc) and having some users try it out.